src/bilder/images/airbyte/deploy.py

```diff
@@ -0,0 +1,134 @@
+from bilder.lib.linux_helpers import DOCKER_COMPOSE_DIRECTORY
+from bridge.lib.magic_numbers import VAULT_HTTP_PORT
+from bridge.lib.versions import CONSUL_VERSION, VAULT_VERSION
+from bridge.secrets.sops import set_env_secrets
+
+TEMPLATES_DIRECTORY = Path(__file__).resolve().parent.joinpath("templates")
+FILES_DIRECTORY = Path(__file__).resolve().parent.joinpath("files")
+VERSIONS = {
+    "consul": os.environ.get("CONSUL_VERSION", CONSUL_VERSION),
+    "vault": os.environ.get("VAULT_VERSION", VAULT_VERSION),
+}
+
+set_env_secrets(Path("consul/consul.env"))
+
+pomerium_config = PomeriumConfig()
+configure_pomerium(pomerium_config)
+
+files.put(
+    name="Place the airbyte docker-compose.yaml file",
+    src=str(Path(__file__).resolve().parent.joinpath("files", "docker-compose.yaml")),
+    dest=DOCKER_COMPOSE_DIRECTORY,
+    mode="0660",
+)
+files.put(
+    name="Place the airbyte .env file",
+    src=str(Path(__file__).resolve().parent.joinpath("files", ".env")),
+    dest=DOCKER_COMPOSE_DIRECTORY,
+    mode="0660",
+)
+
+vault_templates = [
+    VaultTemplate(
+        source=pomerium_config.configuration_template_file,
+        destination=Path(pomerium_config.configuration_file),
+    ),
+    VaultTemplate(
+        contents='{{ with secret "secret-operations/global/odl_wildcard_cert" }}'
+        "{{ printf .Data.key }}{{ end }}",
+        destination=Path(f"{pomerium_config.certificate_key_file}"),
+    ),
+    VaultTemplate(
+        contents='{{ with secret "secret-operations/global/odl_wildcard_cert" }}'
+        "{{ printf .Data.value }}{{ end }}",
+        destination=Path(f"{pomerium_config.certificate_file}"),
+    ),
+]
+consul_configuration = {
+    Path("00-default.json"): ConsulConfig(
+        addresses=ConsulAddresses(dns="127.0.0.1", http="127.0.0.1"),
+        advertise_addr='{{ GetInterfaceIP "ens5" }}',
+    )
+}
+vector_config = VectorConfig(is_proxy=False)
+
+vault_config = VaultAgentConfig(
+    cache=VaultAgentCache(use_auto_auth_token="force"),
+    listener=[
+        VaultListener(
+            tcp=VaultTCPListener(
+                address=f"127.0.0.1:{VAULT_HTTP_PORT}", tls_disable=True
+            )
+        )
+    ],
+    vault=VaultConnectionConfig(
+        address=f"https://vault.query.consul:{VAULT_HTTP_PORT}",
+        tls_skip_verify=True,
+    ),
+    auto_auth=VaultAutoAuthConfig(
+        method=VaultAutoAuthMethod(
+            type="aws",
+            mount_path="auth/aws",
+            config=VaultAutoAuthAWS(role="airbyte-server"),
+        ),
+        sink=[VaultAutoAuthSink(type="file", config=[VaultAutoAuthFileSink()])],
+    ),
+    template=vault_templates,
+)
+vault = Vault(
+    version=VERSIONS["vault"],
+    configuration={Path("vault.json"): vault_config},
+)
+consul = Consul(version=VERSIONS["consul"], configuration=consul_configuration)
+consul_template_config = ConsulTemplate()
+
+hashicorp_products = [vault, consul]
+install_hashicorp_products(hashicorp_products)
```

Comment:
If you're building on top of the docker-compose base image then this step shouldn't be necessary

Reply:
I believe you're correct. I was getting pretty confused at one point with this and what happens in the base AMI and what happens here. I think lines 124:125 do the needful I was actually looking for.